lua/mason-registry/google-java-format/init.lua

```diff
@@ -0,0 +1,33 @@
+local Pkg = require "mason-core.package"
+local github = require "mason-core.managers.github"
+local path = require "mason-core.path"
+
+return Pkg.new {
+    name = "google-java-format",
+    desc = [[google-java-format is a program that reformats Java source code to comply with Google Java Style.]],
+    homepage = "https://github.com/google/google-java-format",
+    languages = { Pkg.Lang.Java },
+    categories = { Pkg.Cat.Formatter },
+    ---@async
+    ---@param ctx InstallContext
+    install = function(ctx)
+        github
+            .download_release_file({
+                repo = "google/google-java-format",
+                out_file = "google-java-format.jar",
+                asset_file = function(release)
+                    local version = release:gsub("^v", "")
+                    return ("google-java-format-%s.jar"):format(version)
```

Comment:
I think it'd be better to target the uber jar, otherwise users would have to ensure dependencies exist in the class path. Or is there some reason to use the slim jar?

```suggestion
                    return ("google-java-format-%s-all-deps.jar"):format(version)
```